blockchain/setup.go

```diff
@@ -1,210 +1,121 @@
+	sdk, err := fabsdk.New(config.FromFile(setup.ConfigFile))
 	if err != nil {
-		return nil, fmt.Errorf("Initialize the config failed: %v", err)
+		return fmt.Errorf("failed to create sdk: %v", err)
 	}
+	setup.sdk = sdk
 
-	// Initialize blockchain cryptographic service provider (BCCSP)
-	// This tool manages certificates and keys
-	err = bccspFactory.InitFactories(configImpl.GetCSPConfig())
+	// Channel management client is responsible for managing channels (create/update channel)
+	// Supply user that has privileges to create channel (in this case orderer admin)
+	chMgmtClient, err := setup.sdk.NewClient(fabsdk.WithUser(setup.OrgAdmin), fabsdk.WithOrg(setup.OrgName)).ChannelMgmt()
 	if err != nil {
-		return nil, fmt.Errorf("Failed getting ephemeral software-based BCCSP [%s]", err)
+		return fmt.Errorf("failed to add Admin user to sdk: %v", err)
 	}
 
-	// This will make a user access (here the admin) to interact with the network
-	// To do so, it will contact the Fabric CA to check if the user has access
-	// and give it to him (enrollment)
-	client, err := fcutil.GetClient("admin", "adminpw", "/tmp/enroll_user", configImpl)
+	// Org admin user is signing user for creating channel.
+	// The session method is the only way for now to get the user identity.
+	session, err := setup.sdk.NewClient(fabsdk.WithUser(setup.OrgAdmin), fabsdk.WithOrg(setup.OrgName)).Session()
 	if err != nil {
-		return nil, fmt.Errorf("Create client failed: %v", err)
+		return fmt.Errorf("failed to get session for %s, %s: %s", setup.OrgName, setup.OrgAdmin, err)
 	}
-	setup.Client = client
+	orgAdminUser := session
 
-	// Make a new instance of channel pre-configured with the info we have provided,
-	// but for now we can't use this channel because we need to create and
-	// make some peer join it
-	channel, err := fcutil.GetChannel(setup.Client, setup.ChannelId)
-	if err != nil {
-		return nil, fmt.Errorf("Create channel (%s) failed: %v", setup.ChannelId, err)
-	}
-	setup.Channel = channel
-
-	// Get an orderer user that will validate a proposed order
-	// The authentication will be made with local certificates
-	ordererUser, err := fcutil.GetPreEnrolledUser(
-		client,
-		"ordererOrganizations/example.com/users/Admin@example.com/keystore",
-		"ordererOrganizations/example.com/users/Admin@example.com/signcerts",
-		"ordererAdmin",
-	)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to get the orderer user failed: %v", err)
-	}
-
-	// Get an organisation user (admin) that will be used to sign the proposal
-	// The authentication will be made with local certificates
-	orgUser, err := fcutil.GetPreEnrolledUser(
-		client,
-		"peerOrganizations/org1.example.com/users/Admin@org1.example.com/keystore",
-		"peerOrganizations/org1.example.com/users/Admin@org1.example.com/signcerts",
-		"peerorg1Admin",
-	)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to get the organisation user failed: %v", err)
+	// Create channel
```

Comment:
Explain in more detail. A quick overview of a channel ?

Reply:
@MaximeAubanel

Reply:
Done